### DISCUSSION TITLE
Job groups that can be scheduled and trigger other groups/jobs

### DISCUSSION THREAD

**Discussion by MrMeeb** [jhuckaby/Cronicle#402] [Ideas]
Hi all. At my current workplace we use a job scheduler called Tivoli, made by IBM. It has some features I feel could be useful to Cronicle, primarily its ability to have a group of jobs, called a 'Job Stream'. This is a collection of jobs that run in sequence (much like how you can chain jobs together in Cronicle), but it feels a bit more complete, especially for Cronicle installs that have a large collection of jobs. 

This could fit into Cronicle by working alongside your existing 'Categories'. Categories could exist as a repository of jobs, which could then be selected and ordered as part of one or multiple 'Job Streams'. This could make reusing a single job in multiple schedules or job chains easier, since currently that job would have to be duplicated in Cronicle, making managing that script more difficult as it would exist across multiple instances. 'Job Streams', like jobs, could also have the ability to be chained together, allowing more complex schedules, such as:

[Job Stream 1: {Job 1 -> Job 2 -> Job 3}] -> [Job Stream 2: {Job 1 -> Job 2 -> Job 3}]

As it is, Cronicle feels like it's designed around the single scheduled job, with some thought given to job chaining. Adding job groups I feel would add to this.

**Comment by jhuckaby** [comment]
Yup, this is coming in Cronicle v2 (a.k.a Orchestra), which is due for release in 2022.  It has "Workflows" which are basically exactly what you are describing: a sequence of jobs, but it has some really neat features where individual jobs can control the flow (skip around, repeat, start other workflows, trigger actions, etc.).

Thanks for the feature request!  Sorry for the delay on this reply.

**Comment by MrMeeb** [comment]
Very good! Can't wait to see what Orchestra has in the pipeline

**Reply by oliviermollier** [reply]
Hello, any news about Orchestra ? Seems so many nice feature?

**Reply by jhuckaby** [reply]
Hey there!  I'm so sorry for lack of updates on Orchestra.  I am working on it every possible free moment I have, and it's coming along quite nicely, but I still have a ways to go.

I apologize -- I keep meaning to create a blog or at least a landing page with Dev updates, possibly alpha / beta releases, and a Twitter account, but life keeps getting in the way.  And also Twitter is a dumpster fire lately, thanks to Elon 😝, so I guess I'll have to look into Mastodon.

I'm just a solo developer on the app, and working on it nights & weekends, so progress is sporadic and unpredictable.  But I am 100% committed to completing and releasing it, without fail.

I missed my 2022 deadline, alas, but 2023 is the year!

**Reply by oliviermollier** [reply]
Hello, i definitly understand :) it's just your software is very great and I think we are all impatient :) Congratulation anyway
